authentication-service/src/main/java/com/innowise/authenticationservice/service/impl/UserServiceImpl.java

```diff
@@ -3,49 +3,86 @@
 import com.innowise.authenticationservice.exception.ResourceAlreadyExistsException;
 import com.innowise.authenticationservice.exception.ResourceNotFoundException;
 import com.innowise.authenticationservice.model.RoleEnum;
-import com.innowise.authenticationservice.model.dto.AuthRequest;
+import com.innowise.authenticationservice.model.dto.RegistrationRequest;
 import com.innowise.authenticationservice.model.dto.UserDto;
 import com.innowise.authenticationservice.model.entity.Role;
 import com.innowise.authenticationservice.model.entity.User;
 import com.innowise.authenticationservice.repository.RoleRepository;
 import com.innowise.authenticationservice.repository.UserRepository;
 import com.innowise.authenticationservice.service.UserService;
+import com.innowise.authenticationservice.service.UserServiceRestClient;
 import com.innowise.authenticationservice.util.ExceptionMessageGenerator;
 import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
+import org.springframework.transaction.support.TransactionSynchronization;
+import org.springframework.transaction.support.TransactionSynchronizationManager;
 
 import java.util.List;
 
 @Service
+@Slf4j
 @RequiredArgsConstructor
 public class UserServiceImpl implements UserService {
 
+    private final UserServiceRestClient userServiceRestClient;
+
     private final UserRepository userRepository;
     private final RoleRepository roleRepository;
 
     private final PasswordEncoder passwordEncoder;
 
     @Override
     @Transactional
-    public UserDto save(AuthRequest authRequest) {
-        if (userRepository.existsByEmail(authRequest.getEmail())) {
-            throw new ResourceAlreadyExistsException(ExceptionMessageGenerator.userExists(authRequest.getEmail()));
+    public UserDto save(RegistrationRequest registrationRequest) {
+        if (userRepository.existsByEmail(registrationRequest.getEmail())) {
+            throw new ResourceAlreadyExistsException(ExceptionMessageGenerator.userExists(registrationRequest.getEmail()));
         }
-
-        User user = User.builder()
-                .email(authRequest.getEmail())
-                .password(passwordEncoder.encode(authRequest.getPassword()))
+        
+        UserDto userServiceDto = UserDto.builder()
+                .name(registrationRequest.getName())
+                .surname(registrationRequest.getSurname())
+                .birthDate(registrationRequest.getBirthDate())
+                .email(registrationRequest.getEmail())
                 .build();
 
+        UserDto userDto = userServiceRestClient.save(userServiceDto);
+
+        prepareRollBackSaveUserInUserService(userDto);
```

Comment:
nah, you previous solution was better
prepate for rollback - its very odd but good try